File: trunk/activeobject/src/scheduler.rb
require "src/activation_queue"

class Scheduler
  def initialize
    @activation_queue = ActivationQueue.new
    Thread.new {
      while true
        @activation_queue.dequeue.call
      end
     }
  end
  def enqueue(&item)
    @activation_queue.enqueue(item)
  end
end

File: trunk/activeobject/java/src/activeobject/Scheduler.java
package activeobject;

public class Scheduler {
	private ActivationQueue<Runnable> activationQueue = new ActivationQueue<Runnable>();

	public Scheduler() {
		Thread thread = new Thread() {
			public void run() {
				while (true) {
					activationQueue.dequeue().run();
				}
			}
		};
		thread.start();
	}

	public void enqueue(Runnable runnable) {
		activationQueue.enqueue(runnable);
	}

}
